### PR TITLE
Upload spdlog package to tket/stable

### PR DESCRIPTION
**Pull request by cqc-alec**
Follows the pattern of the symengine workflow.

This version of `spdlog` differs from the version on conan-center in that it does not have `fmt` as an external dependency.

In a [follow-up PR](https://github.com/CQCL/tket/pull/323) I will make tket depend on this version.

### CODE REVIEW

File: .github/workflows/build_spdlog.yml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      run: |
+        $spdlog_ver = conan inspect --raw version recipes/spdlog/
+        echo "SPDLOG_VER=${spdlog_ver}" >> $env:GITHUB_ENV
+    - name: upload package (dry run)
+      if: github.event_name == 'pull_request'
+      run: conan upload spdlog/${{ env.SPDLOG_VER }}@tket/stable --all -r=tket-conan --skip-upload
+    - name: upload package
+      if: github.event_name == 'push'
+      run: conan upload spdlog/${{ env.SPDLOG_VER }}@tket/stable --all -r=tket-conan
+  manylinux:
```
Comment (cqc-melf) [inline .github/workflows/build_spdlog.yml:168]: Why is the linux and the manylinux build needed?

File: recipes/spdlog/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+sources:
+  "1.10.0":
+    url: "https://github.com/gabime/spdlog/archive/v1.10.0.tar.gz"
+    sha256: "697f91700237dbae2326b90469be32b876b2b44888302afbc7aceb68bcfe8224"
```
Comment (cqc-melf) [inline recipes/spdlog/conandata.yml:4]: Should I check the hash values?

File: .github/workflows/build_spdlog.yml (hunk 1 above)
Comment (cqc-alec) [reply .github/workflows/build_spdlog.yml:168]: The manylinux build is for use by the release builds. It differs in that it uses a different libcpp.

File: recipes/spdlog/conandata.yml (hunk 2 above)
Comment (cqc-alec) [reply recipes/spdlog/conandata.yml:4]: This is copied from https://github.com/conan-io/conan-center-index/blob/master/recipes/spdlog/all/conandata.yml

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**